app/models/relationship.rb

```diff
@@ -26,12 +26,12 @@ class Relationship < ApplicationRecord
   after_update :reset_counters, prepend: true
   after_update :propagate_inversion
   after_save :turn_off_unmatched_field, if: proc { |r| r.is_confirmed? || r.is_suggested? }
-  after_save :move_explainers_to_source, if: proc { |r| r.is_confirmed? }
+  after_save :move_explainers_to_source, :apply_status_to_children, if: proc { |r| r.is_confirmed? }
   before_destroy :archive_detach_to_list
   after_destroy :update_counters, prepend: true
   after_destroy :turn_on_unmatched_field, if: proc { |r| r.is_confirmed? || r.is_suggested? }
   after_commit :update_counter_and_elasticsearch, on: [:create, :update]
-  after_commit :update_counters, :destroy_elasticsearch_relation, on: :destroy
```

Comment:
I removed `update_counters` callback as it's already exists in `after_destroy` L31